### PR TITLE
socket_option: don't set socket IP socket options on pipes.

### PR DESCRIPTION
**Pull request by htuch**
Fixes oss-fuzz issue https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=12293.

Risk level: Low
Testing: Corpus entry and unit test added.

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

File: source/common/network/addr_family_aware_socket_option_impl.cc [hunk 1]
```diff
@@ -11,17 +11,24 @@ namespace Envoy {
 namespace Network {
 
 namespace {
+Address::IpVersion getVersionFromAddress(Address::InstanceConstSharedPtr addr) {
+  if (addr->ip() != nullptr) {
+    return addr->ip()->version();
+  }
+  throw EnvoyException("Unable to set socket option on non-IP sockets");
```
Comment (venilnoronha) [inline source/common/network/addr_family_aware_socket_option_impl.cc]: Should this method throw a more generic `EnvoyException`, followed by `setOption` throwing this one?

Comment (htuch) [reply source/common/network/addr_family_aware_socket_option_impl.cc]: This is caught in `getVersionFromSocket()` and converted to a nullopt return.

Comment (venilnoronha) [reply source/common/network/addr_family_aware_socket_option_impl.cc]: I just felt that the exception message didn't match the function's operation.

Comment (htuch) [reply source/common/network/addr_family_aware_socket_option_impl.cc]: Yeah, technically it doesn't, but I don't think it's a big deal given the hyper local use here.

**Pull request merged**